pkg/kine/drivers/generic/generic.go

```diff
@@ -381,6 +393,119 @@ func (d *Generic) queryRowPrepared(ctx context.Context, txName, sql string, prep
 	return r
 }
 
+func (d *Generic) InitializeWriteBatching(ctx context.Context) {
+	if d.BatchingEnabled && d.BatchingInterval > 0 && d.BatchingMaxQueries > 0 {
+		d.flushCh = make(chan struct{})
+		d.addToBatchQueueCh = make(chan BatchedInsert)
+		d.removeFromBatchQueueCh = make(chan int)
+		d.batchingQueue = make([]BatchedInsert, 0, d.BatchingMaxQueries)
+
+		go d.watchBatchQueue(ctx)
+		go d.batchQueueWatcher(ctx)
+	}
+}
+
+func (d *Generic) batchQueueWatcher(ctx context.Context) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case item := <-d.addToBatchQueueCh:
+			d.batchingQueue = append(d.batchingQueue, item)
+			if len(d.batchingQueue) == d.BatchingMaxQueries {
+				d.flushCh <- struct{}{}
+			}
+		case count := <-d.removeFromBatchQueueCh:
+			d.batchingQueue = d.batchingQueue[count:]
```

Comment:
These most likely need a mutex to protect

Reply:
Just to learn something: Why do we need to protect the queue here? We only change the queue through this watcher and therefore the changes should not happen concurrently, right?